### PR TITLE
Rewrite test app

### PR DESCRIPTION
**Pull request by zLukas**

### CODE REVIEW

**Review by marcin-le — CHANGES_REQUESTED** [review]
application should go to tests/tools

**Comment by zLukas** [comment]
> application should go to tests/tools

done

**Comment by MateuszGrabuszynski** [comment]
Wait a minute, we need to have ONLY changes made specifically for this PR in the PR, without the jiggle-juggle of the other stuff. I rebased val-dev to main, could you please rebase your fork's branch to val-dev? Thanks. :)

**Comment by MateuszGrabuszynski** [comment]
The rebase removed the copyright header changes, but should we consider changes from `.github/workflows` as a part of this PR?

**Comment by zLukas** [comment]
> The rebase removed the copyright header changes, but should we consider changes from `.github/workflows` as a part of this PR?

unnecessary changes cleaned up

File: tests/tools/TestApp/README.md [hunk 1]
```diff
@@ -0,0 +1,32 @@
+#TestApp
```
Comment (KarolinaPomian) [inline tests/tools/TestApp/README.md:1]:
```suggestion
# TestApp
```

Comment (zLukas) [reply tests/tools/TestApp/README.md]: done

**Review by MateuszGrabuszynski — APPROVED** [review]
LGTM 👍🏻

**Pull request merged**